### PR TITLE
Add a yaml sdb module

### PR DESCRIPTION
**Pull request by whiteinge**
### What does this PR do?

Add a new sdb module for loading flat YAML files.

### CODE REVIEW

File: salt/sdb/yaml.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    Setting a value is not supported; edit the YAML files directly
+    '''
+    raise salt.exceptions.NotImplemented()
+
+
+def get(key, profile=None):  # pylint: disable=W0613
+    '''
+    Get a value from the REST interface
+    '''
+    data = _get_values(profile)
```
Comment (rallytime) [inline salt/sdb/yaml.py]: Should we set `profile` to an empty dict here if it's not passed in? I'm just thinking that the `profile.get` call in line 76 will stacktrace with an AttributeErrpr if this is still none.

**Review by rallytime — COMMENTED** [review]
@whiteinge I just had a couple of quick comments here.

**Comment by whiteinge** [comment]
Good feedback. Thanks.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Pull request merged**